### DISCUSSION TITLE
Confusion over domains, launch menu and requesting custom profiles.

### DISCUSSION THREAD

**Discussion by 1player** [wezterm/wezterm#2099] [General]
Here's what I'm trying to achieve: I use an immutable distro called Fedora Silverblue, and all my development is done inside a container generated with `distrobox`. I want to create two desktop files and helper shell commands to either open a Wezterminal into the host, or inside the container toolbox. Ideally both terminals should have a custom background colour to differentiate them, and lastly, that when I create a new tab, it uses the correct profile, i.e. creating a new tab on the toolbox Wezterm should give me a shell inside the toolbox. 

I've tried to wrap my head around about domains, launch menu and whether I can just set a custom Lua variable and have a different configuration for each, but still no joy. 

To recap, my goals are: 

* Have two profiles with their custom shell, background color, color scheme, UNIX socket and spawning a different shell when I create a new tab.
* Being able to pick one or the other at start time, i.e. `wezterm start --profile=profile1`

I've read the documentation for a while, tried something out, but most of my confusion stems from the fact that:

* I don't understand how I could create a UNIX domain inside of my container to connect to. And even if I could, not sure how to set a custom background and shell based on the domain.
* I cannot create a custom configuration for each domain or launch_menu item, such as having a different background colour.
* There doesn't seem to be a way for me to pass Wezterm a custom lua variable and to write all this logic myself. Something like `wezterm --set-variable profile=toolbox`.

Either I'm missing something, or it seems domains, launch menus, profiles are just a few polishes away from being rolled into a "custom profile" concept that is able to change _any_ part of the configuration.

Similar discussions:

* https://github.com/wez/wezterm/discussions/2093
* https://github.com/wez/wezterm/discussions/2081
* https://github.com/wez/wezterm/issues/117

**Comment by wez** [comment]
I'm not sure I understand all of the details for that environment, but what I'd suggest as a first pass is:

* In the container:
  * create a wezterm.lua file that has the colors/colorscheme settings you want to use to differentiate it from the host
  * Assuming that you have a way to map files either from the host into the container, or from the container back to the host, define a unix domain and configure its `socket_path` to whatever that host-visible location is inside the container.
* on the host:
  * create a wezterm.lua file that has the colors/colorscheme you want for the host
  * define a unix domain and configure its `socket_path` to the host-visible location of the path to the container's socket
  * configure the `serve_command` for that unix domain to be something like `serve_command = {"distrobox-enter", "distroname", "--", "wezterm-mux-server", "--daemonize"}`.  That will allow wezterm that is running on the host to be able to enter the container and start the mux server inside it when needed.

Then, when you start wezterm on the host you will by default have a host shell, but if you run `wezterm connect NAME` (where `NAME` is the name you picked for your unix domain), then you will connect into the container instead.

if you wish to invert that, so that the default is to put you inside the container, you can use `default_gui_startup_args = {"connect", "NAME"},` to make that the default, and use `wezterm start` to explicitly start on the host.

When it comes to launch menu items, you can set `domain = {DomainName='NAME'}` or `domain={DomainName='local'}` to explicitly specify where you want those to run if you don't want them to run in the same domain as the current pane.

In the future, I want to do something along the lines of what is discussed in:
* https://github.com/wez/wezterm/issues/1776

so that this sort of workflow can be expressed without having to run the multiplexer server inside the target container (similar to how wsl domains work in wezterm today on windows).

**Comment by 1player** [comment]
I will give this a try, but one thing to note: the point of a toolbox is sharing the same home directory with the container. So I cannot easily have two separate `~/.config/wezterm/wezterm.lua` files, I need to place one somewhere else and start one of the terminals with `wezterm --config-file`, which is another reason why multiple profiles would be preferable, instead of maintaining two separate config files and commands.

**Reply by wez** [reply]
Note that you could use `os.getenv("IS_CONTAINER")` in your wezterm config to adjust configuration based on the environment that you're setting up in https://github.com/wez/wezterm/discussions/2130 but I think the user vars are a nice solution for this sort of thing.
